cqrs-postgres/src/db_wrapper.rs

```diff
@@ -0,0 +1,180 @@
+        since: Since,
+        params: &[Box<dyn ToSql>],
+    ) -> Result<Vec<RawEvent>, Self::Error>;
+}
+
+impl<'conn> DbPool<'conn> for Pool<PostgresConnectionManager> {
+    type Connection = PooledConnection<PostgresConnectionManager>;
+    type Error = DbError;
+
+    fn get(&self) -> Result<Self::Connection, Self::Error> {
+        match self.get() {
+            Ok(c) => Ok(c),
+            Err(err) => Err(DbError::from(err)),
+        }
+    }
+}
+
+impl<'conn> DbConnection<'conn> for PooledConnection<PostgresConnectionManager> {
+    type Error = DbError;
+
+    fn load_since(&self, reaction_name: &str) -> Result<Since, Self::Error> {
+        let stmt = self.prepare_cached(
+            "SELECT event_id \
+             FROM reactions \
+             WHERE reaction_name = $1 \
+             LIMIT 1",
+        )?;
+
+        let rows = stmt.query(&[&reaction_name])?;
+
+        for row in &rows {
+            let event_id: Sequence = row.get("event_id");
+            return Ok(Since::Event(event_id.0));
+        }
+
+        Ok(Since::BeginningOfStream)
+    }
+
+    fn save_since(&self, reaction_name: &str, event_id: EventNumber) -> Result<(), Self::Error> {
+        let stmt = self.prepare_cached(
+            "INSERT INTO reactions (reaction_name, event_id) \
+             VALUES ($1, $2) \
+             ON CONFLICT (reaction_name) \
+             DO UPDATE SET event_id = EXCLUDED.event_id",
+        )?;
+
+        stmt.execute(&[
+            &reaction_name,
+            &event_id
+                .get()
+                .to_i64()
+                .expect("Not expecting event_id > several billions"),
+        ])?;
+
+        Ok(())
+    }
+
+    fn read_all_events(
+        &self,
+        query: &str,
+        since: Since,
+        params: &[Box<ToSql>],
+    ) -> Result<Vec<RawEvent>, Self::Error> {
+        let last_sequence = match since {
+            Since::BeginningOfStream => 0,
+            Since::Event(x) => x.get(),
+        } as i64;
+
+        fn handle_row(row: Row) -> RawEvent {
+            let event_id: Sequence = row.get("event_id");
+            let aggregate_type = row.get("aggregate_type");
+            let entity_id = row.get("entity_id");
+            let sequence: Sequence = row.get("sequence");
+            let event_type = row.get("event_type");
+            let payload = row.get_bytes("payload").unwrap();
+            RawEvent {
+                event_id: event_id.0,
+                aggregate_type,
+                entity_id,
+                sequence: sequence.0,
+                event_type,
+                payload: payload.to_owned(),
+            }
+        };
+
+        let events: Vec<RawEvent>;
+        {
+            let rows = {
+                let local_params: Vec<_> = ::std::iter::once::<&dyn ToSql>(&last_sequence)
+                    .chain(params.iter().map(|p| &**p))
+                    .collect();
+                let stmt = self.prepare_cached(query)?;
+                dbg!(&query);
```

Comment:
Should these `dbg!` calls still be here?

Reply:
lines 169, 170, 175

Reply:
Nope. Clippy didn't think so, either.